File: frontend/src/api/export.ts
import { api } from "./client";

export async function downloadExport(params: {
  country?: string;
  date_from?: string;
  date_to?: string;
  search?: string;
  tag_type?: string;
  tag_value?: string;
  format: "csv" | "xlsx";
}): Promise<void> {
  const response = await api.post("/export", params, {
    responseType: "blob",
  });
  const blob = new Blob([response.data]);
  const url = window.URL.createObjectURL(blob);
  const a = document.createElement("a");
  a.href = url;
  const ext = params.format === "xlsx" ? "xlsx" : "csv";
  a.download = `export.${ext}`;
  document.body.appendChild(a);
  a.click();
  a.remove();
  window.URL.revokeObjectURL(url);
}

File: backend/app/schemas/export.py
from pydantic import BaseModel


class ExportRequest(BaseModel):
    country: str | None = None
    date_from: str | None = None
    date_to: str | None = None
    search: str | None = None
    tag_type: str | None = None
    tag_value: str | None = None
    format: str = "csv"
